bin/list

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env bash
+
+########################################################################################
+#  DEFAULTS
+########################################################################################
+
+USER_DATA_PATH=$(if [[ -d ${JUPYTER_DATA_DIR}/kernels ]]; then
+                echo ${JUPYTER_DATA_DIR}/kernels
+            elif [[ $(uname -s) == Darwin ]]; then
+                echo $HOME/Library/Jupyter/kernels
+            elif [[ -d ${XDG_DATA_HOME}/jupyter/kernels ]]; then
+                echo $XDG_DATA_HOME/jupyter/kernels
+            else
+                echo $HOME/.local/share/jupyter/kernels
+            fi)
+USER_LIB_PATH=$(if [[ $(uname -s) == Darwin ]]; then
+                  echo $HOME/Library
+              else
+                  echo $HOME/.local/lib
+              fi)
+SYSTEM_DATA_PATH=$(if [[ $(uname -s) == Darwin ]]; then
+                  echo /usr/share/jupyter/kernels
+              else
+                  echo /usr/local/share/jupyter/kernels
+              fi)
+SYSTEM_LIB_PATH=$(if [[ $(uname -s) == Darwin ]]; then
+                  echo /usr/lib
+              else
+                  echo /usr/local/lib
+              fi)
+
+########################################################################################
+#  ARGUMENTS & FAIL CONDITIONS
+########################################################################################
+
+all=""
+for dir in $USER_DATA_PATH/*/ $SYSTEM_DATA_PATH/*/; do
+    match=$(jq '.display_name' $dir/kernel.json 2>/dev/null | sed -nE 's/"Clojure \((.*)\)"/\1/g p')
```

Comment:
Can we assume `jq` is installed? We should check for that and bail if not

Reply:
> Can we assume jq is installed?

No, I totally forgot about this. `jq` is indeed a dependecy that won't come preinstalled with most OSs.
We only need it to get the display name in kernel.json so I guess our use case is constrained enough to get away without a general purpose json parser and use some regex instead.

The powershell script doesn't have this issue because you don't need external programs to parse json.